### ISSUE TITLE
Messages are being corrupted on receipt in the client in Unity 2019.

### ISSUE THREAD

**Issue by JamJar00** [DarkRiftNetworking/DarkRift#101]
### Quick Description
Messages are being corrupted on receipt in the client in Unity 2019.

### Explanation
When using Unity 2019 and DR 2.5.0 messages being received on the client can corrupted. Symptoms have been recorded as:
- When sending a string, the reader Length is correct but the data is truncated (Whack)
- Messages received with an invalid tag/message ID as well as very strange message size, e.g. the server would send message 117 and 178 bytes and the client would read 1795 ID and 2626236 bytes. (Aurelius)

When using 2.4.5 in the client the issue is resolved.
No issues have been reported on Unity 2018 yet.

**Comment by Walta69** [comment]
To add my case to this issue (Using DR 2.5.0 with Unity 2019.3.6f1 and not using DR serialization):

* I only experienced this issue when running over LAN or WAN, I could not successfully reproduce this when connecting over localhost.

* I **seem** to only be able to reproduce this with packet sizes over 1400 bytes. The message corruption only seemed to occur once my state packet became too large, around 5 minutes into a game. My Serializer then throws an ArgumentOutOfRangeException when it tries to Deserialize my state packet comprising of lists.

Using DR 2.4.5 fixed the issue.

**Comment by Armynator** [comment]
Tested 2.5.0 over LAN with Unity 2019.3.7f1, data got corrupted and connection was closed randomly when sending a lot of data at once.

I tried something simple like this:
`for (int i = 0; i < 2000; i++) {
writer.Write(i);
}
`
The first few hundred messages are usually going through without issues. But after around 300-700 (results varied) sent integers, the server received wrong values and automatically performed a dirty close of the connection. (Client lost connection, but wasn't disconnected on server side?!) Testing the same code with lower values between 10 and 250 worked fine a few times on 2.5.0.

Could this be related to some cache somehow? It looks like this always happens after a specific amount of data has been sent. Completely restarting the application fixes the issue for the next X amount of messages before it appears again.

Downgraded to 2.4.5 and the issue is gone. I tested 2.4.5 multiple times with even larger amounts of data, same result: no issues with 2.4.5.
